klaus/views.py

```diff
@@ -281,6 +284,54 @@ def make_template_context(self, *args):
         self.context['filename'] = os.path.basename(self.context['path'])
 
 
+class SubmoduleView(BaseRepoView):
+    """Show an information page about a submodule."""
+    template_name = 'submodule.html'
+
+    def make_template_context(self, repo, rev, path):
+        repo, rev, commit = _get_repo_and_rev(repo, rev)
+
+        try:
```

Comment:
I'd like to see the next two code blocks (next 20 lines) factored out into one or more functions/methods.

Not 100% sure what makes sense here but I guess it's useful to at least factor out the "read submodule URL/path from config" part.